boa/src/builtins/value/mod.rs

```diff
@@ -208,37 +208,39 @@ impl Value {
             ValueData::Object(ref obj) => {
                 if obj.borrow().kind == ObjectKind::Array {
                     let mut arr: Vec<JSONValue> = Vec::new();
-                    obj.borrow().properties.keys().for_each(|k| {
+                    for k in obj.borrow().properties.keys() {
                         if k != "length" {
                             let value = self.get_field(k.to_string());
                             if value.is_undefined() || value.is_function() {
                                 arr.push(JSONValue::Null);
                             } else {
                                 arr.push(
-                                    // TODO: Handle the errors
-                                    self.get_field(k.to_string()).to_json(interpreter).unwrap(),
+                                    self.get_field(k.to_string()).to_json(interpreter)?
                                 );
                             }
                         }
-                    });
+                    }
                     Ok(JSONValue::Array(arr))
                 } else {
                     let mut new_obj = Map::new();
-                    obj.borrow().properties.keys().for_each(|k| {
+                    for k in obj.borrow().properties.keys() {
                         let key = k.clone();
                         let value = self.get_field(k.to_string());
                         if !value.is_undefined() && !value.is_function() {
-                            // TODO: Handle the errors
-                            new_obj.insert(key, value.to_json(interpreter).unwrap());
+                            new_obj.insert(key, value.to_json(interpreter)?);
                         }
-                    });
+                    }
                     Ok(JSONValue::Object(new_obj))
                 }
             }
             ValueData::String(ref str) => Ok(JSONValue::String(str.clone())),
-            ValueData::Rational(num) => Ok(JSONValue::Number(
-                JSONNumber::from_f64(num).expect("Could not convert to JSONNumber"),
-            )),
+            ValueData::Rational(num) => {
+                if let Some(number) = JSONNumber::from_f64(num) {
+                    Ok(JSONValue::Number(number))
+                } else {
+                    Ok(JSONValue::Null)
+                }
```

Comment:
I would use this:
```suggestion
                JSONNumber::from_f64(num).map(JsonValue::Number).unwrap_or(JSONValue::Null)
```

Reply:
The suggestion wasn't quite right, but I got it updated.